src/lib.rs

```diff
@@ -55,33 +53,54 @@ pub fn run(config: Config) -> Vec<ViolationWithContext> {
     let tree_sitter_grep_args = get_tree_sitter_grep_args(&aggregated_queries);
     let all_violations: Mutex<HashMap<PathBuf, Vec<ViolationWithContext>>> = Default::default();
     let files_with_fixes: AllPendingFixes = Default::default();
-    let instantiated_per_file_rules: AllInstantiatedPerFileRules = Default::default();
```

Comment:
So this got rid of the whole "use hash buckets + `Mutex`'s to lazily populate per-file-rule-instance's" thing